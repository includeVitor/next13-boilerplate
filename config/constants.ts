enum ApiRoutes {
    posts = 'http://localhost:3000/api/posts',
    post = 'http://localhost:3000/api/posts'
}

enum AppAssets {
    favicon = '/favicon.ico',
    profile = '/profile.jpg'
}

enum AppSegments {
    posts = 'posts'
}

enum AppRoutes {
    home = '/'
}

export { ApiRoutes, AppAssets, AppSegments, AppRoutes }
