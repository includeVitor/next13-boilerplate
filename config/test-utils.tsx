// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import { render, RenderOptions } from "@testing-library/react";
import { FC, ReactElement } from "react";

type ProvidersProps = {
    children: React.ReactNode;
};

const Providers: FC<ProvidersProps> = ({ children }) => {
    return <>{children}</>;
};

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, "queries">
) => render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from "@testing-library/react";
// override render method
export { customRender as render };
