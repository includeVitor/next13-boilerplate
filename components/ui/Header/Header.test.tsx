import { Header } from '@components/ui'
import { AppSegments } from '@config/constants'
import { render } from '@config/test-utils'

jest.mock('next/navigation', () => ({
    _esModule: true,
    useSelectedLayoutSegment: jest
        .fn()
        .mockImplementationOnce(() => AppSegments.posts)
        .mockImplementationOnce(() => null)
}))

describe('Header', () => {
    it('Should be able to render header inside posts segment', () => {
        const { container } = render(<Header />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('Should be able to render header outside posts segment', () => {
        const { container } = render(<Header />)
        expect(container.firstChild).toMatchSnapshot()
    })
})
