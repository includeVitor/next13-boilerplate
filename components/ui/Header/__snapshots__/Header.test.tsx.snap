// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should be able to render header inside posts segment 1`] = `
<header>
  <a
    class="flex flex-col items-center justify-center"
    href="/"
  >
    <img
      alt="profile pic"
      class="rounded-full cursor-pointer"
      data-nimg="1"
      decoding="async"
      height="108"
      loading="lazy"
      src="/_next/image?url=%2Fprofile.jpg&w=256&q=75"
      srcset="/_next/image?url=%2Fprofile.jpg&w=128&q=75 1x, /_next/image?url=%2Fprofile.jpg&w=256&q=75 2x"
      style="color: transparent;"
      width="108"
    />
    <h2
      class="pt-4 pb-6 text-3xl hover:underline cursor-pointer font-bold"
    >
      Username
    </h2>
  </a>
</header>
`;

exports[`Header Should be able to render header outside posts segment 1`] = `
<header
  class="flex flex-col items-center justify-center"
>
  <img
    alt="profile pic"
    class="rounded-full"
    data-nimg="1"
    decoding="async"
    height="144"
    loading="lazy"
    src="/_next/image?url=%2Fprofile.jpg&w=384&q=75"
    srcset="/_next/image?url=%2Fprofile.jpg&w=256&q=75 1x, /_next/image?url=%2Fprofile.jpg&w=384&q=75 2x"
    style="color: transparent;"
    width="144"
  />
  <h1
    class="pt-4 pb-6 text-4xl font-extrabold"
  >
    Username
  </h1>
</header>
`;
